pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -260,7 +260,13 @@ func doCleanSubPaths(mounter mount.Interface, podDir string, volumeName string)
 
 		// scan /var/lib/kubelet/pods/<uid>/volume-subpaths/<volume>/<container name>/*
 		fullContainerDirPath := filepath.Join(subPathDir, containerDir.Name())
-		err = filepath.Walk(fullContainerDirPath, func(path string, info os.FileInfo, _ error) error {
+		// The original traversal method here was ReadDir, which was not so robust to handle some error such as "stale NFS file handle",
+		// so it was replaced with filepath.Walk in a later patch, which can pass through error and handled by the callback WalkFunc.
+		// After go 1.16, WalkDir was introduced, it's more effective than Walk because the callback WalkDirFunc is called before
+		// reading a directory, so if there are lots of dirs in a subPath of a container, it will still traverse all the dirs before
+		// call WalkFunc(here is doCleanSubPath) when using Walk, which is time-wasting and meaningless.
```

Comment:
> which is time-wasting and meaningless.
I feel that this is a little bit too negative 

> so if there are lots of dirs in a subPath of a container, it will still traverse all the dirs before call WalkFunc(here is doCleanSubPath) when using Walk, which is time-wasting and meaningless.
Probably this is redundant

Reply:
Thanks! I have simplified it.